Source/WTF/wtf/cocoa/UUIDCocoa.mm

```diff
@@ -35,11 +35,14 @@
     return [[NSUUID alloc] initWithUUIDString:toString()];
 }
 
-// Use string instead of bytes to avoid consideration of endianess (NSUUID stores UUID as array of bytes,
-// and we store UUID as UInt128).
-UUID::UUID(NSUUID * nsUUID)
-    : UUID(*parse(String([nsUUID UUIDString])))
+std::optional<UUID> UUID::fromNSUUID(NSUUID *nsUUID)
 {
+    if (!nsUUID)
+        return std::nullopt;
+
+    // Use string instead of bytes to avoid consideration of endianess (NSUUID stores UUID as array of bytes,
+    // and we store UUID as UInt128).
+    return parse(String([nsUUID UUIDString]));
```

Comment:
I think you also need to add an explicit check inside parse so that parsing fails for emptyValue. I guess we need an API test for that too.